contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1239 @@
+
+        underlyingAssets = _wethAmount;
+        emit UnderlyingAssetsUpdated(_wethAmount);
+    }
+
+    /**
+     * Burns any OETHp tokens remaining on the strategy contract
+     */
+    function _burnOethOnTheContract() internal {
+        uint256 _oethpBalance = IERC20(OETHp).balanceOf(address(this));
+        if (_oethpBalance > MIN_LIQUIDITY_THRESHOLD) {
+            IVault(vaultAddress).burnForStrategy(_oethpBalance);
+        }
+    }
+
+    /**
+     * @notice Returns the percentage of WETH liquidity in the configured ticker
+     *         owned by this strategy contract.
+     * @return uint256 1e18 denominated percentage expressing the share
+     */
+    function getWETHShare() external view returns (uint256) {
+        uint256 _currentPrice = getPoolSqrtPrice();
+        return _getWethShare(_currentPrice);
+    }
+
+    /**
+     * @dev Returns the share of WETH in tick denominated in 1e18
+     */
+    function _getWethShare(uint256 _currentPrice)
+        internal
+        view
+        returns (uint256)
+    {
+        (
+            uint256 wethAmount,
+            uint256 oethpAmount
+        ) = _reservesInTickForGivenPriceAndLiquidity(
+                sqrtPriceTickLower,
+                sqrtPriceTickHigher,
+                _currentPrice,
+                1e24
+            );
+
+        return wethAmount.divPrecisely(wethAmount + oethpAmount);
+    }
+
+    /**
+     * @notice Returns the current pool price in square root
+     * @return Square root of the pool price
+     */
+    function getPoolSqrtPrice() public view returns (uint256) {
+        return poolLens.getPoolSqrtPrice(mPool);
+    }
+
+    /**
+     * @notice Returns the current active trading tick of the underlying pool
+     * @return _currentTick Current pool trading tick
+     */
+    function getCurrentTradingTick() public view returns (int32 _currentTick) {
+        _currentTick = mPool.getState().activeTick;
+    }
+
+    /**
+     * @notice Mint the initial NFT position
+     *
+     * @dev This amount is "gifted" to the strategy contract and will count as a yield
+     *      surplus.
+     */
+    // slither-disable-start reentrancy-no-eth
+    function mintInitialPosition() external onlyGovernor nonReentrant {
+        require(tokenId == 0, "Initial position already minted");
+        (
+            bytes memory packedSqrtPriceBreaks,
+            bytes[] memory packedArgs,
+            ,
+            uint256 OETHpRequired
+        ) = _getAddLiquidityParams(1e16, 1e16);
+
+        // Mint rounded up OETH amount
+        if (OETHpRequired > 0) {
+            IVault(vaultAddress).mintForStrategy(OETHpRequired);
+        }
+
+        _approveTokenAmounts(1e16, OETHpRequired);
+
+        (, , , uint256 _tokenId) = liquidityManager.mintPositionNftToSender(
+            mPool,
+            packedSqrtPriceBreaks,
+            packedArgs
+        );
+        // burn remaining OETHp
+        _burnOethOnTheContract();
```

Comment:
⚪ Sometimes we burnOethOnTheContract() before we update the _updateUnderlyingAssets(), and sometimes after. Would be cleaner to be consistent. I think it would be cleanest to burn before, but either way should work.